ArduPlane/parachute.cpp

```diff
@@ -56,3 +59,62 @@ bool Plane::parachute_manual_release()
 
     return true;
 }
+
+// Code to detect loss of control for ArduPlane
+#define PARACHUTE_CHECK_TRIGGER_SEC         1  // 1 second of loss of control triggers the parachute
+
+/*
+  parachute_emergency_check - trigger the release of the parachute
+  automatically if a critical situation is detected
+*/
+void Plane::parachute_emergency_check()
+{
+    static uint32_t control_loss_ms;  // time when we first lost control and never regained it
+    uint32_t now = millis();
+
+    // exit immediately if parachute or automatic release is not enabled, or already released
+    if (!parachute.enabled() || parachute.released() || g.parachute_auto_enabled == 0) {
+        control_loss_ms = 0;
+        return;
+    }
+
+    // only automatically release in AUTO mode
+    if (control_mode != AUTO) {
+        control_loss_ms = 0;
+        return;
+    }
+
+    // do not release if vehicle is not flying
+    if (!is_flying()) {
```

Comment:
OK. I think it should be safe to remove this check, because we also have the `relative_altitude` check below, which ensures that we are in the air.

Also, since `Plane::parachute_manual_release()` also prevents parachute deployment when not `is_flying()` maybe it should be removed from there as well? @tridge